Chap_API_Publish.tex

```diff
@@ -542,59 +591,28 @@ \subsubsection{Lookup data structure support macros}
-
-%%%%
-\summary
-
-The \refapi{pmix_lookup_cbfunc_t} is used by \refapi{PMIx_Lookup_nb} to return data.
-
-\copySignature{pmix_lookup_cbfunc_t}{1.0}{
-typedef void (*pmix_lookup_cbfunc_t) \\
-\hspace*{4\sigspace}(pmix_status_t status, \\
-\hspace*{5\sigspace}pmix_pdata_t data[], size_t ndata, \\
-\hspace*{5\sigspace}void *cbdata);
-}
-
-\begin{arglist}
-\argin{status}{Status associated with the operation (handle)}
-\argin{data}{Array of data returned (\refstruct{pmix_pdata_t})}
-\argin{ndata}{Number of elements in the \argref{data} array (\code{size_t})}
-\argin{cbdata}{Callback data passed to original API call (memory reference)}
-\end{arglist}
-
-
-%%%%
-\descr
-
-A callback function for calls to \refapi{PMIx_Lookup_nb}.
-The function will be called upon completion of the \refapi{PMIx_Lookup_nb} \ac{API} with the \refarg{status} indicating the success or failure of the request.
-Any retrieved data will be returned in an array of \refstruct{pmix_pdata_t} structs.
-The namespace and rank of the process that provided each data element is also returned.
-
-Note that the \refstruct{pmix_pdata_t} structures will be released upon return from the callback function, so the receiver must copy/protect the data prior to returning if it needs to be retained.
-
-
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \section{Retrieval rules for published data}
 \label{chap:pub:retrules}
 
-The retrieval rules for published data primarily revolve around enforcing data access permissions and range constraints. The datastore shall search its stored information for each specified key according to the following precedence logic:
+The retrieval rules for published data primarily revolve around enforcing data access permissions and range constraints.
+All publish and lookup operations operate on a range. If not specified, the range defaults to \refconst{PMIX_RANGE_SESSION}.
+The key being looked up will match with a published key only if all of the following conditions are met:
 
 \begin{enumerate}
-    \item If the requester specified the range, then the search shall be constrained to data where the publishing process falls within the specified range.
-
-    \item If the key of the stored information does not match the specified key, then the search will continue.
-
-    \item If the requester's identifier does not fall within the range specified by the publisher, then the search will continue.
-
-    \item If the publisher specified access permissions, the effective \ac{UID} and \ac{GID} of the requester shall be checked against those permissions, with the datastore rejecting the match if the requester fails to meet the requirements.
-
-    \item If all of the above checks pass, then the value is added to the information that is to be returned.
+    \item The lookup key matches the published key.
+    \item The type of range specified by the publisher is the same as the type of range specified by the requester.
```

Comment:
Is this required (e.g., requestor specifies 'node' range, and publisher specifies 'global' range)? Or is it enough that the requestor be part of the publisher's range?

Reply:
The new model is that the ranges must match... if you publish to 'global', then the requestor must specifically lookup on 'global' range.